src/scene_graph/scene_graph.hpp

```diff
@@ -128,6 +128,8 @@ namespace wr
 		std::vector<std::shared_ptr<LightNode>> m_light_nodes;
 		std::vector< std::shared_ptr<SkyboxNode>> m_skybox_nodes;
 
+		std::shared_ptr<SkyboxNode> m_default_skybox = nullptr;
```

Comment:
the default skybox should probably be part of the scene graph tree itself for consistent behavior.

Reply:
Default skybox is a "fallback" for when a user deletes the last skybox in the scene tree. Adding it to the tree would allow its deletion which is not expected behaviour. Due to the descriptor heap system we're using, something NEEDS to be bound in the skybox slots even if the shaders are not going to sample from it, the default skybox has a 2x2 cubemap that is used as a fallback in case there's not skybox nodes available at render time